subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/SwiftXCTestErrorHandlingIntegrationTest.groovy

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest
+
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.integtests.fixtures.TestExecutionResult
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.RequiresInstalledToolChain
+import org.gradle.nativeplatform.fixtures.ToolChainRequirement
+import org.gradle.nativeplatform.fixtures.app.SwiftAppWithLibrariesAndXCTest
+import org.gradle.nativeplatform.fixtures.app.XCTestCaseElement
+import org.gradle.nativeplatform.fixtures.app.XCTestSourceElement
+import org.gradle.nativeplatform.fixtures.app.XCTestSourceFileElement
+
+import static org.gradle.util.Matchers.containsText
+
+@RequiresInstalledToolChain(ToolChainRequirement.SWIFTC)
+class SwiftXCTestErrorHandlingIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
+
+    def "fails when application cannot load shared library at runtime"() {
+        buildWithApplicationAndDependencies()
+        buildFile << """
+            project(':app') {
+                tasks.withType(XCTest).configureEach {
+                    doFirst {
+                        delete project(':hello').layout.buildDir.get()
+                    }
+                }
+            }
+        """
+
+        expect:
+        fails(':app:test')
+
+        and:
+        failure.assertHasCause("There were failing tests.")
```

Comment:
This is better, but still a little strange. In the report, I get a test called `Gradle Test Run :app:xcTest. execution failure`.  We may be constrained with how much we can change this without breaking other things, but maybe label this as `Gradle Test Run :app:xcTest. failed to start test process`

Reply:
I'm not sure I agree.  `failed to start test process` implies well, that it failed to start the test process, which is not the case.  It started the test process, it's just that the test process experienced a fatal error.  Would `failed to execute tests` be clearer?

Reply:
I've update this to "failed to execute tests".  I don't mind changing it to something better, but "failed to start test process" is probably a little over-specific.